### PR TITLE
Hui Hui CSS Battle

### PR DESCRIPTION
**Pull request by hui444**
* Add css battles 1-12, 93, 36 and 95
* Add animation example mentioned previously

### CODE REVIEW

**Comment by hui444** [comment]
[Example card](https://trello.com/c/7mSUajFx/2-example-card)

File: src/CSSBattle/1_SimplySquare/HuiHui.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<div class="box"></div>
+<style>
+  body {
+    background: #5d3a3a;
+    margin: 0;
+  }
+
+  .box {
+    width: 200px;
+    aspect-ratio: 1;
```
Comment (xinyee20) [inline src/CSSBattle/1_SimplySquare/HuiHui.html:10]: Good use of aspect-ratio to control the size of the box, a square in this case

File: src/CSSBattle/3_PushButton/HuiHui.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<div class="btn"></div>
+
+<style>
+  body {
+    background: #6592cf;
+    margin: 0;
+    display: grid;
```
Comment (xinyee20) [inline src/CSSBattle/3_PushButton/HuiHui.html]: My initial thought was that grid is more useful when there are many items in a 2D layout and that flexbox is more suitable in this case where there is just one big shape in the middle of the layout but interestingly you manage to use grid to centralize the shapes in 2 lines. That saved a line more! Compared to when using flexbox (display: flex; justify-content: center; align-items: center;) Nice try at minimizing code

File: src/CSSBattle/3_PushButton/HuiHui.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    place-items: center;
+  }
+
+  .btn {
+    background: #eeb850;
+    width: 50px;
+    height: 50px;
+    border-radius: 50%;
+    border: 50px solid #243d83;
+    box-shadow: 0 0 0 50px #6592cf;
```
Comment (xinyee20) [inline src/CSSBattle/3_PushButton/HuiHui.html:17]: Smart way to use `border` and `box-shadow` to create the 3 perfect rings of circle

File: src/CSSBattle/3_PushButton/HuiHui.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  .btn::after {
```
Comment (xinyee20) [inline src/CSSBattle/3_PushButton/HuiHui.html:20]: Very interesting and pretty unconventional way to add the box behind. An alternative and maybe more direct way to do the same is to add a new div before the div with class="btn" which has a class modified from .btn::after :
.outerbox {
    width: 300px;
    height: 150px;
    background: #243d83;
    position: absolute;
  }

File: src/CSSBattle/3_PushButton/HuiHui.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<div class="btn"></div>
+
+<style>
+  body {
+    background: #6592cf;
+    margin: 0;
```
Comment (xinyee20) [inline src/CSSBattle/3_PushButton/HuiHui.html]: can be removed

File: src/CSSBattle/3_PushButton/HuiHui.html [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    display: grid;
+    place-items: center;
```
Comment (xinyee20) [inline src/CSSBattle/3_PushButton/HuiHui.html]: Seems like iframe has some alignment issues. I tried modifying your code to use flexbox instead and the same problem arise, until I remove align-items and set a margin-top of 75px then the image stays in the middle of the iframe.

File: src/CSSBattle/4_UpsNDowns/HuiHui.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+<style>
+  body {
+    background: #62306d;
+    margin: 0;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin-top: -100px;
+  }
```
Comment (xinyee20) [inline src/CSSBattle/4_UpsNDowns/HuiHui.html:12]: Does having both margin and margin-top make them conflict? Testing with your code, margin-top: -92px; gives the same effect on cssbattle

File: src/CSSBattle/5_AcidRain/HuiHui.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
+<div class="a"></div>
+
+<style>
+  body {
+    background: #0b2429;
+    display: flex;
+    flex-direction: row;
```
Comment (xinyee20) [inline src/CSSBattle/5_AcidRain/HuiHui.html]: Ultimately there's only one div being used so this line can be removed

File: src/CSSBattle/5_AcidRain/HuiHui.html [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    left: -60px;
+    border-top-right-radius: 0;
+  }
+
+  .a::before {
+    top: -60px;
+    right: -60px;
+    z-index: -1;
+  }
+</style>
```
Comment (xinyee20) [inline src/CSSBattle/5_AcidRain/HuiHui.html:41]: Very nice approach to this challenge!:blush:

File: src/CSSBattle/8_ForkingCrazy/HuiHui.html [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+  .stick,
+  .stem {
+    width: 20px;
+    height: 110px;
+    background: #060f55;
+    border-radius: 50px;
+  }
+
+  .stick:nth-child(2n) {
```
Comment (xinyee20) [inline src/CSSBattle/8_ForkingCrazy/HuiHui.html:32]: Sick infusion of math :astonished:

File: src/CSSBattle/10_CloakedSpirits/HuiHui.html [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  .b {
+    width: 60px;
+    height: 60px;
+    background: #aa445f;
+    border-radius: 50%;
+    border: 20px solid #e38f66;
+  }
+
+  .a:after,
+  .b::after {
```
Comment (xinyee20) [inline src/CSSBattle/10_CloakedSpirits/HuiHui.html:33]: Interesting creating the cloak using ::after. Are you missing a ':' in line 32?

File: src/CSSBattle/11_EyeOfSauron/HuiHui.html [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    height: 50px;
+    background: #84271c;
+    border-radius: 50%;
+    border: 25px solid #191210;
+    box-shadow: 0 0 0 20px #eca03d;
+    position: relative;
+  }
+
+  .circle::before,
+  .circle::after {
```
Comment (xinyee20) [inline src/CSSBattle/11_EyeOfSauron/HuiHui.html:21]: Smart way to create perfect semicircles:+1:

File: src/CSSBattle/36_Interleaved/HuiHui.html [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    background: #1a4341;
+    display: flex;
+    gap: 25px;
+    margin: 0 25px;
+  }
+
+  .stick {
+    width: 50px;
+    height: 300px;
+    background: #dd6b4d;
```
Comment (xinyee20) [inline src/CSSBattle/36_Interleaved/HuiHui.html]: Can be removed since is redeclared below

File: src/CSSBattle/93_GreatWall/HuiHui.html [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+    place-items: center;
+    background: #4f77ff;
+  }
+
+  .big-circle {
+    width: 200px;
+    height: 200px;
+    background: #dd6b4d;
+    border-radius: 50%;
+    background: #191919;
```
Comment (xinyee20) [inline src/CSSBattle/93_GreatWall/HuiHui.html:19]: Line 18 and 20 conflict, remove one

**Review by xinyee20 — CHANGES_REQUESTED** [review]
A few minor stuff to change but good job for coming up with quite a few interesting solutions to the challenges! :smile: [Review 1]

File: src/CSSBattle/10_CloakedSpirits/HuiHui.html (hunk 11 above)
Comment (hui444) [reply src/CSSBattle/10_CloakedSpirits/HuiHui.html:33]: good catch! fixed in ed24aa2

File: src/CSSBattle/36_Interleaved/HuiHui.html (hunk 13 above)
Comment (hui444) [reply src/CSSBattle/36_Interleaved/HuiHui.html]: fixed in 4f7a70a

File: src/CSSBattle/3_PushButton/HuiHui.html (hunk 2 above)
Comment (hui444) [reply src/CSSBattle/3_PushButton/HuiHui.html]: `place-item` is a shorthand for both `justify-items` and `align-items` so works for grid and not flexbox haha learnt it recently!

File: src/CSSBattle/3_PushButton/HuiHui.html (hunk 4 above)
Comment (hui444) [reply src/CSSBattle/3_PushButton/HuiHui.html:20]: haha thats true! maybe was too focused on using pseudo classes

File: src/CSSBattle/3_PushButton/HuiHui.html (hunk 6 above)
Comment (hui444) [reply src/CSSBattle/3_PushButton/HuiHui.html]: tested and works! 0349d6e

File: src/CSSBattle/3_PushButton/HuiHui.html (hunk 5 above)
Comment (hui444) [reply src/CSSBattle/3_PushButton/HuiHui.html]: fixed in c7f487f

File: src/CSSBattle/4_UpsNDowns/HuiHui.html (hunk 7 above)
Comment (hui444) [reply src/CSSBattle/4_UpsNDowns/HuiHui.html:12]: nope, second line overwrites first. but updated to use 1 liner in 71b5359
not too sure about changing margin-top to -92px, tried on css battle and does not give the same results?

File: src/CSSBattle/93_GreatWall/HuiHui.html (hunk 14 above)
Comment (hui444) [reply src/CSSBattle/93_GreatWall/HuiHui.html:19]: fixed in ae91892

File: src/CSSBattle/5_AcidRain/HuiHui.html (hunk 8 above)
Comment (hui444) [reply src/CSSBattle/5_AcidRain/HuiHui.html]: good catch! fixed in db8626f

**Comment by Magmanat** [comment]
I need to learn to be as good as you, Review 2: LGTM

File: src/CSSBattle/3_PushButton/HuiHui.html (hunk 4 above)
Comment (xinyee20) [reply src/CSSBattle/3_PushButton/HuiHui.html:20]: pseudo classes are pretty cool!

**Review by xinyee20 — APPROVED** [review]
lgtm!

File: src/CSSBattle/4_UpsNDowns/HuiHui.html (hunk 7 above)
Comment (xinyee20) [reply src/CSSBattle/4_UpsNDowns/HuiHui.html:12]: My apologies, yes I think -100px makes more sense

**Review by Magmanat — APPROVED** [review]
I need to learn to be as good as you, Review 2: LGTM

**Pull request merged**